src/Raven.Client/Json/Serialization/JsonNet/Internal/JsonNetBlittableEntitySerializer.cs

```diff
@@ -1,50 +1,46 @@
 ﻿using System;
-using System.Threading;
-using Newtonsoft.Json;
 using Newtonsoft.Json.Linq;
+using Raven.Client.Documents.Conventions;
 using Raven.Client.Documents.Identity;
-using Raven.Client.Json;
 using Sparrow.Json;
 using Sparrow.Threading;
-using Sparrow.Utils;
 
-namespace Raven.Client.Documents.Conventions
+namespace Raven.Client.Json.Serialization.JsonNet.Internal
 {
     internal class JsonNetBlittableEntitySerializer
     {
-        private readonly LightWeightThreadLocal<BlittableJsonReader> _reader;                
-        private readonly LightWeightThreadLocal<JsonSerializer> _deserializer;
+        private readonly LightWeightThreadLocal<BlittableJsonReader> _reader;
+        private readonly LightWeightThreadLocal<IJsonSerializer> _deserializer;
 
+        private readonly Lazy<GenerateEntityIdOnTheClient> _generateEntityIdOnTheClient;
 
-        private readonly GenerateEntityIdOnTheClient _generateEntityIdOnTheClient;
-
-        public JsonNetBlittableEntitySerializer(DocumentConventions conventions)
+        public JsonNetBlittableEntitySerializer(ISerializationConventions conventions)
         {
-            _generateEntityIdOnTheClient = new GenerateEntityIdOnTheClient(conventions, null);            
-            _deserializer = new LightWeightThreadLocal<JsonSerializer>(conventions.CreateDeserializer);
+            _generateEntityIdOnTheClient = new Lazy<GenerateEntityIdOnTheClient>(() => new GenerateEntityIdOnTheClient(conventions.Conventions, null));
```

Comment:
why lazy?

Reply:
This is a problematic part. The 'DocumentConventions' might be in ctor.  Not sure how to make it convenient for the user.

I would like them to be able to do:

```
Conventions
{
    Serialization = new JsonNetSerializationConventions
    {
        // ...
    }
}
```

and I did that, but this means that I cannot require in ctor of JsonNetSerializationConventions the parent DocumentConventions.

Reply:
You can set it when in the `Serialization` setter, no?

Reply:
Not sure that I follow?

Reply:
User writes:

```
Conventions
{
    Serialization = new JsonNetSerializationConventions
    {
        // ...
    }
}
```

And inside we do:

```
Conventsion.Serialization
{
   set { 
          _serialization = value;
          _serialization.Initiaize(this);
   }
}
```

So here we provide the right value

Reply:
Ok, will check.